### PR TITLE
Add Product class and field classes

### PR DESCRIPTION
**Pull request by DavidTan0527**

### CODE REVIEW

File: src/main/java/seedu/address/model/product/Category.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    /**
+     * A {@code Category} singleton representing no categorization.
+     */
+    public static final Category NONE = new Category();
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Categories should only contain alphanumeric characters and spaces";
+
+    /*
+     * The first character of the address must not be a whitespace,
```
Comment (1001mei) [inline src/main/java/seedu/address/model/product/Category.java:21]: Should this be first character of category instead? Found this typo in several other places also.

File: src/main/java/seedu/address/model/product/Name.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package seedu.address.model.product;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's name in the address book.
```
Comment (1001mei) [inline src/main/java/seedu/address/model/product/Name.java:7]: Should this be Product's name? Found this issue in Price.java also.

**Review by 1001mei — COMMENTED** [review]
LGTM. Just found some typos in documentation.

**Comment by DavidTan0527** [comment]
Build failed due to the Storage component not being updated to adapt to AddressBook using Product.
I will revert back to the old AddressBook and create a new class.

**Pull request merged**